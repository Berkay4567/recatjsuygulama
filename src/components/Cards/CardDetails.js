import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

import styles from "./Cards.module.scss";
import { Link } from "react-router-dom";
const CardDetails = () => {
    let { id } = useParams();
  
    let [fetchedData, updateFetchedData] = useState([]);
    let { name, job, avatar, about } = fetchedData;
  
    let api = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons/${id}`;
  
    useEffect(() => {
      (async function () {
        let data = await fetch(api).then((res) => res.json());
        updateFetchedData(data);
      })();
    }, [api]);
  
    return (
      <div className="container d-flex justify-content-center mb-5">
        <div className="d-flex flex-column gap-3">
          <h1 className="text-center">{name}</h1>
          <img className="img-fluid" src={avatar} alt="" />
          <div className="content">
            <div className="">
              <span className="fw-bold">Job : </span>
              {job}
            </div>
            <div className="">
              <span className="fw-bold">About: </span>
              {about}
            </div>
            <br/>
            <Link className="mt-3 w-100 text-center" to="/">
						<button className={styles.button2} >Geri DÃ¶n</button>
			</Link>
          </div>
        </div>
      </div>
    );
  };

export default CardDetails