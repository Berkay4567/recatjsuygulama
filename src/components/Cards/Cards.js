import React from 'react';
import styles from "./Cards.module.scss";
import { Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "bootstrap/dist/js/bootstrap";
const Cards = ({page, fetchedData}) =>{
    let display;
    console.log(fetchedData);

    if(fetchedData){
        display=fetchedData.map(x=>{
            let {id, name,avatar}=x;

            return(
                <Link
                style={{ textDecoration: "none" }}
                to={`${page}${id}`}
                key={id}
                className="col-lg-4 col-md-6 col-sm-6 col-12 mb-4 position-relative text-dark"
              >
                    <div className={styles.cards}>
                        <img src={avatar} alt="" className={`${styles.img} img-fluid`} />
                        <div className={{padding: "20px"}}>
                            <div className="fs-4 fw-bold mb-4">{name}</div>
                        </div>
                        <button className={styles.button} onClick={()=>sil(id)} to="/"> SİL </button>
                    </div>
                </Link>
            )
            //-------------SİLME KOMUTU--------------------------
            function sil(id){
                try{
                    return fetch(`https://5fc9346b2af77700165ae514.mockapi.io/simpsons/${id}`,
                     { method: 'DELETE' }
                     );
                }
                catch(err){
                    console.log(err);
                }
            }
            //-------------SİLME KOMUTU BİTİMİ-------------------
        });
    }else{
        display="";
    }
    return <>{display}</>;
};

export default Cards