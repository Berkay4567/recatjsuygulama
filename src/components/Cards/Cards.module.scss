$radius: 10px;
.cards {
    border: 2px solid #4877bd;
    border-radius: $radius;
  }

  .img {
    border-radius: $radius $radius 0 0;
  }
  .button{
    display: inline-block;
    padding: 3px ;
    width: calc(100% - 0px);
    font-size: 24px;
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    outline: none;
    color: #fff;
    background-color: #882929;
    border: none;
    border-radius: 0 0 15px 15px;
    box-shadow: 0 9px #999;
    top: 50%;
    left: 50%;
}
.button2{
    padding: 15px 62px;
    font-size: 24px;
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    outline: none;
    color: #fff;
    background-color: #295a88;
    border: none;
    border-radius:15px;
    box-shadow: 0 9px #999;
    text-align: center;
}