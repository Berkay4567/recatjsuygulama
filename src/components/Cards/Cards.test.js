import React from 'react';
import renderer from 'react-test-renderer'
import { BrowserRouter as Router} from 'react-router-dom';
import Cards from './Cards';

it("renders when there are no items", ()=>{
    const tree=renderer.create(<Router>
        <Cards />
      </Router>
  ).toJSON();
    expect(tree).toMatchSnapshot();
})