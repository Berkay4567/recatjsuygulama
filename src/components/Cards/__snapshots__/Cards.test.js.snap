// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no irems 1`] = `""`;

exports[`renders when there are no items 1`] = `""`;
